Add boolean detect

shnotsh opened the pull request:
Added detection for boolean types. Some users may use spreadsheets to calculate formulas that produce boolean outputs. Typically, when exporting, these values are written as "TRUE" or "FALSE." I have implemented a case-insensitive solution in case users manually enter boolean fields in their CSV files using different capitalization conventions.

shnotsh committed:
Add boolean detect

shnotsh committed:
Fix missing quote in README.md

timothyqiu commented on addons/csv-data-importer/import_plugin.gd:
```diff
@@ -93,15 +94,25 @@ func _import(source_file, save_path, options, platform_variants, gen_files):
+				if options.detect_numbers and field.is_valid_float(): # Any valid integer is also a valid float
+					if not options.force_float and field.is_valid_int():
+						detected.append(int(field))
+						continue
+					elif field.is_valid_float():
```
```suggestion
						continue
					if field.is_valid_float():
```

`elif` is not necessary since `continue` returns immediately.

The same applies to the `elif` below too.

timothyqiu commented on addons/csv-data-importer/import_plugin.gd:
```diff
@@ -93,15 +94,25 @@ func _import(source_file, save_path, options, platform_variants, gen_files):
+						continue
+				if options.detect_booleans:
+					if field.nocasecmp_to("false") == 0: 
+						detected.append(bool(false))
```
```suggestion
						detected.append(false)
```

`false` is already boolean.

The same applies to the `true` below too.

shnotsh committed:
Minor fixes

shnotsh commented:
Thanks. Addressed suggestions you mentioned. `is_valid_float()` check is actually unnecessary at all, since it's already validated to be a float earlier.

timothyqiu commented on addons/csv-data-importer/import_plugin.gd:
```diff
@@ -93,15 +94,23 @@ func _import(source_file, save_path, options, platform_variants, gen_files):
-					detected.append(field)
+					continue
+				if options.detect_booleans:
+					if field.nocasecmp_to("false") == 0: 
```
```suggestion
					if field.nocasecmp_to("false") == 0:
```

Trailing space :)

Looks good otherwise.

shnotsh committed:
Remove trailing space

Co-authored-by: Haoyu Qiu <timothyqiu@users.noreply.github.com>

timothyqiu commented:
Thanks!
